py/async_cli/damas_async.py

```diff
@@ -0,0 +1,446 @@
+                      self.serverURL,
+                      "CONNECTION OK --------------------------------"
+                      )
+                self.response = response
+                if callback is None:
+                    print("there is no callback provided")
+                    return self.response
+                else:
+                    res_callback = callback(self.response)
+                    print("the {} callback is executed".format(callback.__name__))
+                    if res_callback is None:
+                        print("The call back return none. \nReturned value before callback is {}".format(self.response))
+                        return self.response
+                    return res_callback
+
+    async def disconnect(self):
+        await self.session.close()
+
+    def set_new_session(self, loop):
+        self.session = aiohttp.ClientSession(loop=loop)
+
+    async def create(self, keys, callback=None):
+        try:
+            assert (keys != {})
+            assert (keys is not None)
+        except EmptyElementException:
+            raise EmptyElementException(keys)
+        async with aiohttp.ClientSession(loop=self.loop) as session:
+            self.session = session
+            if self.session.closed:
+                print("session is closed !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!")
+                raise ClosedSessionExecption
+            url = self.serverURL + "/api/create/"
+            headers = {'content-type': 'application/json'}
+            headers.update(self.headers)
+            data = json.dumps(keys)
+            async with session.post(url, data=data, headers=headers, ssl=False) as resp:
+                assert resp is not None
+                if resp.status < 300:
+                    json_result = await resp.json()
+                    try:
+                        assert (json_result is not None)
+                    except NoneResultExecption:
+                        NoneResultExecption().throw("Create return None")
+                    if callback is None:
+                        print("There is no callback provided")
+                        print(json_result)
+                        return json_result
+                    else:
+                        print("callback given is {}".format(callback.__name__))
+                        callback_result = callback(json_result)
+                        if callback_result is None:
+                            print("The call back return none. \nReturned value before callback is {}".format(self.response))
+                            return self.response
+                    return callback_result
+                else:
+                    if resp.status == 409:
+                        print("Code http create request error is {}".format(resp.status))
+                        raise AlreadyExistException(keys)
+                    print("Code http create request error is {}".format(resp.status))
```

Comment:
these prints are more for your own debug while writing the client library than for a production use case